### ISSUE TITLE
tiling.c doesn't compile on older distributions with gcc7

### ISSUE THREAD

**Issue by cryptomilk** [darktable-org/darktable#7605]
**Describe the bug**

The merge request #7575 breaks building darktable on with older compiler versions.

```
[  407s] /home/abuild/rpmbuild/BUILD/darktable-3.5.0~git298.601bc99a2/src/develop/tiling.c: In function '_default_process_tiling_ptp':
[  407s] /home/abuild/rpmbuild/BUILD/darktable-3.5.0~git298.601bc99a2/src/develop/tiling.c:758:13: error: 'ioffs' is predetermined 'shared' for 'shared'
[  407s]        shared(input, width, ioffs) \
[  407s]              ^
[  407s] /home/abuild/rpmbuild/BUILD/darktable-3.5.0~git298.601bc99a2/src/develop/tiling.c: In function '_default_process_tiling_roi':
[  407s] /home/abuild/rpmbuild/BUILD/darktable-3.5.0~git298.601bc99a2/src/develop/tiling.c:1107:13: error: 'ioffs' is predetermined 'shared' for 'shared'
[  407s]        shared(input, ioffs, iroi_full) \
[  407s]              ^
[  407s] /home/abuild/rpmbuild/BUILD/darktable-3.5.0~git298.601bc99a2/src/develop/tiling.c: In function '_default_process_tiling_cl_ptp':
[  407s] /home/abuild/rpmbuild/BUILD/darktable-3.5.0~git298.601bc99a2/src/develop/tiling.c:1425:15: error: 'ioffs' is predetermined 'shared' for 'shared'
[  407s]          shared(input_buffer, width, ioffs, wd, ht) \
[  407s]                ^
[  407s] /home/abuild/rpmbuild/BUILD/darktable-3.5.0~git298.601bc99a2/src/develop/tiling.c:1425:15: error: 'wd' is predetermined 'shared' for 'shared'
[  407s] /home/abuild/rpmbuild/BUILD/darktable-3.5.0~git298.601bc99a2/src/develop/tiling.c:1425:15: error: 'ht' is predetermined 'shared' for 'shared'
[  407s] /home/abuild/rpmbuild/BUILD/darktable-3.5.0~git298.601bc99a2/src/develop/tiling.c: In function '_default_process_tiling_cl_roi':
[  407s] /home/abuild/rpmbuild/BUILD/darktable-3.5.0~git298.601bc99a2/src/develop/tiling.c:1883:15: error: 'ioffs' is predetermined 'shared' for 'shared'
[  407s]          shared(input_buffer, width, ioffs, iroi_full) schedule(static)
[  407s]                ^
[  407s] /home/abuild/rpmbuild/BUILD/darktable-3.5.0~git298.601bc99a2/src/develop/tiling.c:1932:15: error: 'ooffs' is predetermined 'shared' for 'shared'
[  407s]          shared(ooffs, output_buffer, oroi_full, oorigin, oregion) \
[  407s]                ^
```

See https://build.opensuse.org/package/show/graphics:darktable:master/darktable

**Comment by cryptomilk** [comment]
@ralfbrown

**Comment by ralfbrown** [comment]
Yeah, I noticed this morning -- but I didn't add the 'const's on `ioffs`/`ooffs` that break the OpenMP directives for GCC7/8.

Fixed it as part of rebasing #7587.

**Comment by ralfbrown** [comment]
@TurboGit The consts you added in db0446816 caused the compile errors.  You can cherry-pick the changes to tiling.c in https://github.com/darktable-org/darktable/pull/7587/commits/b79e5880f60f8e851b6329e467167e3d8b2f6048 to fix.

**Comment by ralfbrown** [comment]
@cryptomilk Should be fixed now, so please test & close.

**Comment by johnny-bit** [comment]
Tested on CI and added GCC7 to CI in #7626

I'll close this because I think It is indeed fixed (and CI runs of #7626 indicate no problems with consts and OpenMP :) )
